### DISCUSSION TITLE
Migrate to Docker Compose V2

### DISCUSSION THREAD

**Discussion by Timelessprod** [apache/airflow#39429] [Ideas]
Hello dear Airflow community,

Since July 2023 Docker Compose V1 has stopped receiving updates and Docker suggests to migrate to the V2 on their [website](https://docs.docker.com/compose/migrate/). Most importantly, there has been no security updates on V1 since May 10th 2021.

The migration usually requires minor adjustment from my experience with other projects I work on. Indeed the migration mainly consists of replacing `docker-compose` commands with `docker compose`, as well as removing the `version` key in the `docker-compose.yaml` files as it is now useless. The full list of changes to apply during migration is [here](https://docs.docker.com/compose/migrate/).

Would it be possible to consider upgrading to Docker Compose V2? If so, is it already planned?

I saw this PR https://github.com/apache/airflow/pull/28605 which was supposed to do that already, but it was only in a doc file and I can still see a lot of V1 commands in the code currently on master.

I can consider opening a PR for it if it is considered as a good idea :)

Thanks in advance for your further responses.

**Comment by Taragolis** [comment]
> Would it be possible to consider upgrading to Docker Compose V2? If so, is it already planned?

[Airflow **Sample** Docker Compose](https://airflow.apache.org/docs/apache-airflow/stable/howto/docker-compose/index.html#fetching-docker-compose-yaml) file intends to be run in Docker Compose V2 since Airflow 2.8

**Reply by Taragolis** [reply]
> but it was only in a doc file and I can still see a lot of V1 commands in the code currently on master.

> I can consider opening a PR for it if it is considered as a good idea :)

Feel free
